File: OOPDraw/Rectangle.cs
using Nakov.TurtleGraphics;

namespace OOPDraw
{
    public class Rectangle : Shape
    {
        //Properties
        private float Width { get; set; }
        private float Height { get; set; }

        //The 'Constructor'
        public Rectangle(float xOrigin, float yOrigin, float width, float height) : base(xOrigin, yOrigin)
        {
            Width = width;
            Height = height;
        }

        public override void Draw()
        {
            ResetTurtle();
            for (int i = 0; i < 2; i++)
            {
                Turtle.Forward(Height);
                Turtle.Rotate(90);
                Turtle.Forward(Width);
                Turtle.Rotate(90);
            }
        }

        public override void Resize(float x, float y)
        {
            Width = x;
            Height = y;
        }
    }
}

File: OOPDraw VB/OOPDraw/Rectangle.vb
Imports Nakov.TurtleGraphics

Public Class Rectangle
    Inherits Shape
    'Properties
    Private Property Width As Single
    Private Property Height As Single

    'The 'Constructor'
    Public Sub New(xOrigin As Single, yOrigin As Single, width As Single, height As Single)
        MyBase.New(xOrigin, yOrigin)
        Me.Width = width
        Me.Height = height
    End Sub

    Public Overrides Sub Draw()
        ResetTurtle()
        For i = 1 To 10
            Turtle.Forward(Height)
            Turtle.Rotate(90)
            Turtle.Forward(Width)
            Turtle.Rotate(90)
        Next
    End Sub

    Public Overrides Sub Resize(x As Single, y As Single)
        Width = x
        Height = y
    End Sub

End Class
